indico/web/client/js/react/components/ICSCalendarLink.jsx

```diff
@@ -41,44 +50,67 @@ export default function ICSCalendarLink({endpoint, urlParams, options, ...restPr
     }
   };
 
-  const trigger = (
-    <Dropdown
-      {...restProps}
-      icon={null}
-      trigger={
-        <Button icon size="small">
-          <Icon name="calendar alternate outline" />
-          <Icon name="caret down" />
-        </Button>
-      }
-      pointing="top right"
-    >
-      <Dropdown.Menu>
-        <Dropdown.Header>
-          <Translate>Synchronise with your calendar</Translate>
-        </Dropdown.Header>
-        <Dropdown.Divider />
-        {options.map(({key, text, queryParams}) => (
-          <Dropdown.Item
-            key={key}
-            text={text}
-            onClick={async () => {
-              setOption({
-                text,
-                url: await fetchURL(queryParams),
-              });
-              setOpen(true);
-            }}
-          />
-        ))}
-      </Dropdown.Menu>
-    </Dropdown>
-  );
+  const handleSetOption = async (text, queryParams) => {
+    setOption({
+      text,
+      url: await fetchURL(queryParams),
+    });
+    setOpen(true);
+  };
+
+  let trigger;
+
+  if (options.length > 1) {
+    trigger = (
+      <Dropdown
+        styleName="height-full"
+        icon={null}
+        trigger={
+          renderButton ? (
+            renderButton()
```

Comment:
This doesn't trigger any warnings/errors since we end up doing `{...props}` with props being `undefined`?

Reply:
It should not, React always returns an empty object even if props are undefined: https://codepen.io/javfg/pen/jOVGGNv

Reply:
I was more worried about ths jsx unpacking, ie this:

```jsx
const foo = undefined;
const test = <Bar {...foo} />;
```

Reply:
Indeed. While I was at it, I removed the spread of props and just passed onClick, as that is considered cleaner.